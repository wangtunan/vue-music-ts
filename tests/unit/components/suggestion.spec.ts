import { mount, Wrapper, config } from '@vue/test-utils'
config.showDeprecationWarnings = false
import Suggestion from '@/components/suggestion/index.vue'

describe('suggestion.vue', () => {
  let wrapper: Wrapper<Vue>
  beforeEach(() => {
    wrapper = mount(Suggestion)
  })
  it('match snaphots', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('render empty when resultList is empty', async () => {
    expect(wrapper.find('.suggestion-empty').exists()).toBe(true)
    wrapper.setData({
      resultList: [1, 2, 3]
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.suggestion-empty').exists()).toBe(false)
    expect(wrapper.find('ul').isVisible()).toBe(true)
    wrapper.setData({
      resultList: []
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.suggestion-empty').exists()).toBe(true)
    expect(wrapper.find('ul').isVisible()).toBe(false)
  })
  it('change keyword and emit onKeywordChange function', async () => {
    const methods = {
      getSuggestionList: jest.fn()
    }
    wrapper = mount(Suggestion, {
      methods: methods
    })
    // 搜索不为空时，触发事件
    wrapper.setProps({
      keyword: '张宇'
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.props('keyword')).toBe('张宇')
    expect(methods.getSuggestionList).toBeCalledTimes(1)
    // 搜索为空时，不触发事件
    wrapper.setProps({
      keyword: ''
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.props('keyword')).toBe('')
    expect(methods.getSuggestionList).toBeCalledTimes(1)
  })
  it('render resultList success', () => {
    const resultList = [1, 2, 3]
    wrapper = mount(Suggestion, {
      data () {
        return {
          resultList: resultList
        }
      }
    })
    expect(wrapper.findAll('.suggestion-item').length).toBe(resultList.length)
  })
})