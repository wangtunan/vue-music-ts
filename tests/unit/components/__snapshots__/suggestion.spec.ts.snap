// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`suggestion.vue match snaphots 1`] = `
<div class="suggestion-list">
  <ul style="display: none;"></ul>
  <div class="m-empty suggestion-empty"><img src="" alt="" class="image">
    <p class="title">抱歉，暂无数据</p>
  </div>
</div>
`;
